### ISSUE TITLE
Can't autoformat max line length (editorconfig?)

### ISSUE THREAD

**Issue by xenoterracide** [pinterest/ktlint#1665]
```
exceeded max line length (120) (cannot be auto-corrected)
```

```ini
root = true
[*]
indent_size = 2
trim_trailing_whitespace = true
insert_final_newline = true
max_line_length = 120
[*.sh]
end_of_line = lf
[*.bat]
end_of_line = crlf
```

line in question

```kt
private fun extractUsing(divsionDtoMap: Map<String, BusinessDivisionDto>): (BusinessDivisionDto) -> BusinessDivsionEntity {
```

https://github.com/pinterest/ktlint/issues/251#issuecomment-1266545614

```
com.pinterest.ktlint:ktlint-core:0.43.2
com.pinterest.ktlint:ktlint-reporter-baseline:0.43.2
com.pinterest.ktlint:ktlint-reporter-checkstyle:0.43.2
com.pinterest.ktlint:ktlint-reporter-html:0.43.2
com.pinterest.ktlint:ktlint-reporter-json:0.43.2
com.pinterest.ktlint:ktlint-reporter-plain:0.43.2
com.pinterest.ktlint:ktlint-reporter-sarif:0.43.2
com.pinterest.ktlint:ktlint-ruleset-experimental:0.43.2
com.pinterest.ktlint:ktlint-ruleset-standard:0.43.2
com.pinterest.ktlint:ktlint-ruleset-test:0.43.2
com.pinterest:ktlint:0.43.2
```

**Comment by paul-dingemans** [comment]
A signature like this will only be fixed with the `function-signature` rule which is available in the `experimental` ruleset. For example when running CLI command (version 0.47.1) `ktlint -F --experimental` the output is as follows:
```
private fun extractUsing(
  divsionDtoMap: Map<String, BusinessDivisionDto>,
): (BusinessDivisionDto) -> BusinessDivsionEntity {
}
```
